### ISSUE TITLE
Easy Question about PSN/Download Games

### ISSUE THREAD

**Issue by pirsir76** [aldostools/webMAN-MOD#1003]
Hey,

I have a quick question here about webMAN-MOD and downloadable PSN games.
I'm running out of internal space on my PS3 but I have plenty of space on my NAS.
Is it possible to store and launch PSN/Download PS3 games from my NAS using webMAN-MOD?

Thanks

**Comment by aldostools** [comment]
Yes, you can extract the packages to GAMEI folder in the shared of ps3netsrv and launch them through ★ app_home/PS3_GAME or ★ PlayStation®3 

The external games also can be launched from the GAMEI folder stored in USB storage devices in FAT32.

You must enable the app_home and GAMEI options in the section "Scan for content" of /setup.ps3
![image](https://github.com/aldostools/webMAN-MOD/assets/974897/9c511655-e5fb-45df-9571-7e7103653b2c)

The game folder in GAMEI must be named with the game id or the content id of the package.

If you enable the external game data using SELECT+SQUARE you can install the PKG directly on the USB drive (FAT32).
It also works for DLC and game updates. However, the games containing files larger than 4GB will not be playable due the file size limitation of FAT32.

>VERY IMPORTANT: Some PSN games are NOT compatible with GAMEI launched from the debug icon app_home and cause a **black screen**. The main reason of this issue is that some games look for content in /dev_hdd0/game instead of use the relative game path.

**Comment by pirsir76** [comment]
Thanks for the information Aldo.
I'll try this out and get back to you.

**Comment by pirsir76** [comment]
Okay, I think we are going to have to go through this step-by-step sorry Aldo.

"...you can extract the packages to GAMEI folder in the shared of ps3netsrv"
This is the ps3netsrv I have: https://www.myqnap.org/product/ps3netsrv/
It might be a bit different to the one you are using, but there is a link to the github page: https://github.com/aldostools/webMAN-MOD/wiki/~-PS3-NET-Server

"GAMEI folder in the shared of ps3netsrv"
I can't see any GAMEI folder in the shared.
I'm also not too sure what you mean by "shared".
I have a folder that I access through:  PS3 webMAN [Setup]/Files/net0 (192.168.233:38008)
Is that the folder that you are calling "shared"?
There is no "GAMEI" folder in it, it only contains files and folders that I put there myself.

EDIT added question:
Is there a way I can copy files from  [Setup]/Files/net0 (192.168.233:38008) to  [Setup]/Files/dev_hdd0?

**Comment by aldostools** [comment]
The "shared" folder is the remote directory in the PC that ps3netsrv uses as root directory.

GAMEI folder must be created manually i the "shared" folder, so you should see it in PS3 webMAN [Setup]/Files/net0
The path in PS3 browser is http://localhost/net0/GAMEI

You can copy files from any remote folder if you first mount the remote folder as /dev_bdvd (like you do when you mount a game in folder format). Once the remote folder is mounted, you can use the file manager to copy the files from /dev_bdvd

The same concept is used to install packages from /net0/PKG. The remote folder "PKG" is mounted as /dev_bdvd and Package Manager install the files from /dev_bdvd

I don't have any afiliation with the site "myqnap". That site is linked only because it's supposed to use the code from my project. However, I do not support any application external to my project because I don't know if they have made modifications.

**Comment by pirsir76** [comment]
"The "shared" folder is the remote directory in the PC that ps3netsrv uses as root directory.

GAMEI folder must be created manually i the "shared" folder, so you should see it in PS3 webMAN [Setup]/Files/net0
The path in PS3 browser is http://localhost/net0/GAMEI"

Okay, I think I understand that part.

"You can copy files from any remote folder if you first mount the remote folder as /dev_bdvd (like you do when you mount a game in folder format). Once the remote folder is mounted, you can use the file manager to copy the files from /dev_bdvd"

This is new to me because in the past I have just been going to the disc image in the shared folder itself and pressing the X button on the file size on the disc. I have found that it will mount PS3 games and copy PS2 games to dev_hdd0 so I can play them.

I just figured out how to mount the GAMEI with your advice and by experimenting: open remote folder you want to mount (in this case I am going to the "GAMEI" folder I created in the shared) and you should then see up top: /net0/GAMEI, click on the "GAMEI" to mount the remote folder.
Now, the GAMEI folder should be accessible through the /dev_bdvd folder.

All that worked thanks, and now I can extract .pkg files to the PS3 remotely.
I have another question you may be able to help with. 
I still have not figured out yet how to play PSX games remotely. I tried just then to mount the PSXISO folder I have in the shared folder and it mounted correctly, but when I went to play a game by clicking on the file size of the .cue file, it just unmounted and did nothing. I could be doing something wrong.

Can you help with that Aldo?

Thanks

**Comment by aldostools** [comment]
PS2 games are a special case. The ISO is always copied locally to PS3 because PS2 games only work from /dev_hdd0 

The PSX must be a proper backup in CUE+BIN (sectors size 2352) and placed in PSXISO folder.
Multi-BIN format is not supported. All the BIN files or BIN+MP3 must be merged into a single BIN file.

Some PSXISO require PPF patches or select ps1_netemu instead of the default ps1_emu.

**Comment by pirsir76** [comment]
I don't know if you got my question there?
How can I mount/load PSX titles remotely from my NAS?

"The PSX must be a proper backup in CUE+BIN"
All of my PSX are CUE+BIN.

"Multi-BIN format is not supported."
All of my PSX that are Muli-BIN have been joined using CDmage.

All of the PSX I have are working 100% when loaded from PS3 webMAN [Setup]/Files/dev_hdd0/PSXISO. It's just that I can't figure out how to load them remotely from the NAS. I would like to load them from the NAS because of lack of space on my PS3 internal SSD.

**Comment by aldostools** [comment]
To mount remote games (including PSXISO) you first need to have configured the remote IP in /setup.ps3:
https://github.com/aldostools/webMAN-MOD/wiki/~-PS3-NET-Server#webman-mod-settings

Then scan the content using prepISO or Refresh XML and Reload XMB (or SELECT+L3 then reload VSH)

**Comment by pirsir76** [comment]
Thanks for the response Aldo.
I will leave this issue open for now and get back to you when I've managed to mount a PSX game remotely.

**Comment by pirsir76** [comment]
Hello again Aldo,

"To mount remote games (including PSXISO) you first need to have configured the remote IP in /setup.ps3:
https://github.com/aldostools/webMAN-MOD/wiki/~-PS3-NET-Server#webman-mod-settings"
I think I have that part set up correctly. 

"Then scan the content using prepISO or Refresh XML and Reload XMB (or SELECT+L3 then reload VSH)"
This here I think is a problem, when I Refresh XML and Reload XMB it comes up with 0 results.
I'm not sure what I done, but I remember a long time ago disabling this because I thought it was clogging up my menus. (Like it was loading 100s of games to the XMB and I thought it was annoying, so I disabled it.)
Can you help me to enable it?
I have supplied 2 pictures of my "Setup" page that might help as I suspect the problem is one of the settings in there.

![DSC01141](https://github.com/aldostools/webMAN-MOD/assets/57454066/4f4e668a-db3e-494c-8519-85c67e277f72)
![DSC01142](https://github.com/aldostools/webMAN-MOD/assets/57454066/e74f34f4-c76d-4425-8563-cf610907b75a)

**Comment by aldostools** [comment]
Do you have your PS1 games in the PSXISO folder in the shared directory of ps3netsrv?

If you navigate to http://0/net0/PSXISO in the PS3 Internet Browser the games should be listed there.
If they are not listed, webMAN MOD will be unable to detect them.
